src/backend/commands/tablecmds.c

```diff
@@ -16756,12 +16756,11 @@ ATPExecPartExchange(AlteredTableInfo *tab, Relation rel, AlterPartitionCmd *pc)
 		pn = RelationBuildPartitionDesc(rel, false);
 		pcols = get_partition_attrs(pn);
 
+		Assert(pid);
```

Comment:
Why is this check here?

Reply:
Even if `get_part_rule()` is called with `bExistError = bMustExist = true`, it can return `NULL`:
https://github.com/arenadata/gpdb/blob/215d8d8f9fec39300e4eb84977ade4fce040ebc1/src/backend/cdb/cdbpartition.c#L5208-L5209